headers/validation.h

```diff
@@ -108,6 +109,52 @@ auto operator|(T1 /*unused*/, T2 /*unused*/) {
 
 enum Separator { Space, Newline };
 
+//this contains some specific code which would be available in c++20
+namespace C20 {
```

Comment:
Probably nicer to call it something like `cpp20`, (or maybe even `future` as in python).


We could even consider putting this behind an `#ifdef [some cpp20 define]`, but macros are ugly so simpler to keep as is.